torch/nn/parallel/distributed.py

```diff
@@ -108,6 +109,19 @@ class _DDPUnevenInputsConfig(NamedTuple):
     ddp_join_enabled: bool
     ddp_join_divide_by_initial_world_size: bool
 
+# Add a DDPSink to queue call back of out-most backward/graph task,
+# this helps call back is fired after all gradients' calculation
+# is completed.
+class DDPSink(Function):
```

Comment:
Can we test this change when DDP model outputs a tuple (a, b) of tensors? I'm afraid we may hit the issue in https://github.com/pytorch/pytorch/issues/55509, but I hope I'm wrong.